src/Illuminate/Database/Connectors/ConnectionFactory.php

```diff
@@ -39,7 +40,7 @@ public function __construct(Container $container)
      * @param  string|null  $name
      * @return \Illuminate\Database\Connection
      */
-    public function make(array $config, $name = null)
+    public function make(array $config, $name)
```

Comment:
`$name` shouldn't accept `null` since `parseConfig()` requires `$name` only to be `string`.

Reply:
`parseConfig()` is only restricted via doc-block documentation. Since the property isn't really strictly typed, it actually works fine by using `$factory->make($config)` on userland code.

Reply:
For multi-tenant applications where you may want to change the default connections, this might not be the best usage:

![CleanShot 2024-01-02 at 10 24 39](https://github.com/laravel/framework/assets/172966/451227ea-fcef-490d-932f-0060b70e9f8f)

Reply:
sorry I didn't see your reply here, unfortunately I don't get notifications on the Laravel repository anymore (GitHub bug).

Whether you change or not the default connection in that script it's irrelevant since the configuration is being explicitly passed to the `$factory->make()`. The variable you hold is the one that is relevant to which database connection you have.

the `getName()` method is returning exactly what was provided when constructing the connection (`null`), which is my point about this breaking change. It works as intended and it doesn't seem relevant to break it